calyx/opt/src/passes/profiler_instrumentation.rs

```diff
@@ -263,8 +263,99 @@ fn static_group(comp: &mut ir::Component, sigs: &ir::LibrarySignatures) {
 
     // TODO: create probes for structural enables, cell invokes, and primitive invokes
 
-    let group_name_assign_and_cell =
-        create_assignments(comp, sigs, &group_names, None, None, None);
+    // groups to groups that they enabled
+    let mut structural_enable_map: CallsFromGroupMap<calyx_ir::StaticTiming> =
+        HashMap::new();
+    // groups to cells (from non-primitive components) that they invoked
+    let mut cell_invoke_map: CallsFromGroupMap<calyx_ir::StaticTiming> =
+        HashMap::new();
+    // groups to primitives that they invoked
+    let mut primitive_invoke_map: CallsFromGroupMap<calyx_ir::StaticTiming> =
+        HashMap::new();
+
+    for group_ref in comp.static_groups.iter() {
+        let group = &group_ref.borrow();
+        // set to prevent adding multiple probes for a combinational primitive enabled by the group
+        let mut comb_primitives_covered = HashSet::new();
+        let mut primitive_vec: Vec<(Id, ir::Guard<calyx_ir::StaticTiming>)> =
+            Vec::new();
+        for assignment_ref in group.assignments.iter() {
+            let dst_borrow = assignment_ref.dst.borrow();
+            if let ir::PortParent::Group(parent_group_ref) = &dst_borrow.parent
+            {
+                if dst_borrow.name == "go" {
+                    // found an invocation of go
+                    let invoked_group_name =
+                        parent_group_ref.upgrade().borrow().name();
+                    let guard = *(assignment_ref.guard.clone());
```

Comment:
style nit: I prefer `(*assignment_ref.guard).clone();`. This is equivalent but I think makes it easier to see what is going on without looking at the precise types. Since cloning a box (heap allocated val) and then immediately moving it to the stack reads as a little off. Though chances are the compiler would turn it into this anyway.